### PR TITLE
Created the initial express-handlebars stuff

### PR DESCRIPTION
**Pull request by NouraldinS**
modified package.json, comma error
configured the express-handlebars in the ./src/app.js
configured a route that uses the main layout with partials body, footer, head and navbar
RELATES #40 Initialize handlebars

### CODE REVIEW

**Review by yahyaHB — APPROVED** [review]

**Review by yahyaHB — APPROVED** [review]
that works just fine

**Review by sultanassi95 — APPROVED** [review]
reviewed, and sumbitted

**Review by Alaa-Khattab — CHANGES_REQUESTED** [review]
solve the conflict !

**Comment by NouraldinS** [comment]
@Alaa-Khattab Solved!

File: package.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
     "express-handlebars": "^3.0.0",
     "path": "^0.12.7",
     "pg": "^7.4.1",
-    "serve-favicon": "^2.4.5",
+    "serve-favicon": "^2.4.5"
```
Comment (mohanagy) [inline package.json:15]: i don't see that u using this module in your project

File: src/app.js [hunk 2]
```diff
@@ -1,10 +1,19 @@
...
 const app = express();
 
 app.use(express.static(path.join(__dirname, '..', 'public')));
+app.set('views', path.join(__dirname, 'views'));
+app.set('view engine', 'hbs');
+app.engine('hbs', exphbs({
+  extname: 'hbs',
+  layoutsDir: path.join(__dirname, 'views', 'layouts'),
+  partialsDir: path.join(__dirname, 'views', 'partials'),
+  defaultLayout: 'main'
```
Comment (mohanagy) [inline src/app.js:15]: add a semicolon after `main`

File: src/app.js [hunk 3]
```diff
@@ -1,10 +1,19 @@
 const express = require('express');
 const path = require('path');
 const routes = require('./controllers/mainController');
+const exphbs = require('express-handlebars');
```
Comment (mohanagy) [inline src/app.js:4]: absolute  requires should  come before relative requires 
` const exphbs = require('express-handlebars');  `
 ` const routes = require('./controllers/mainController');`

File: src/controllers/mainController.js [hunk 4]
```diff
@@ -4,13 +4,14 @@ const dbCon = require('../database/dbConnection');
...
 router.get('/', (req, res) => {
-  dbCon.query('select * from projects', (dbConError, selectProjectsResult) => {
-    if (dbConError) {
-      return res.status(500).send({ error: dbConError });
-    }
-    return res.send({ result: selectProjectsResult });
-  });
-  // return res.send('Hello world');
+  res.render('login', {});
+//   dbCon.query('select * from projects', (dbConError, selectProjectsResult) => {
```
Comment (mohanagy) [inline src/controllers/mainController.js:8]: remove commented lines

**Review by mohanagy — CHANGES_REQUESTED** [review]

File: src/controllers/mainController.js (hunk 4 above)
Comment (sultanassi95) [reply src/controllers/mainController.js:8]: this route should render the home page that is going to be designed later.
not the login page.
><"

**Review by sultanassi95 — CHANGES_REQUESTED** [review]
route "/" is supposed to render the homepage not the login page. hmmm

File: src/controllers/mainController.js (hunk 4 above)
Comment (mohanagy) [reply src/controllers/mainController.js:8]: its useless now  
so no need for it

**Pull request closed without merging**